hocon/src/main/scala/com/avsystem/commons/hocon/ConfigCompanion.scala

```diff
@@ -0,0 +1,69 @@
+package com.avsystem.commons
+package hocon
+
+import com.avsystem.commons.meta.MacroInstances
+import com.avsystem.commons.misc.ValueOf
+import com.avsystem.commons.serialization.{GenCodec, GenKeyCodec, GenObjectCodec, HasGenObjectCodecWithDeps}
+import com.avsystem.commons.serialization.GenCodec.ReadFailure
+import com.typesafe.config.{Config, ConfigFactory, ConfigObject}
+
+import scala.concurrent.duration.*
+
+trait CommonsHoconCodecs {
+  implicit final val configCodec: GenCodec[Config] = GenCodec.nullable(
+    input =>
+      input.readCustom(ConfigValueMarker).map {
+        case obj: ConfigObject => obj.toConfig
+        case v => throw new ReadFailure(s"expected a config OBJECT, got ${v.valueType}")
+      }.getOrElse {
+        ConfigFactory.parseString(input.readSimple().readString())
+      },
+    (output, value) =>
+      if (!output.writeCustom(ConfigValueMarker, value.root)) {
+        output.writeSimple().writeString(value.root.render)
+      },
+  )
+
+  implicit final val finiteDurationCodec: GenCodec[FiniteDuration] = GenCodec.nullable(
+    input => input.readCustom(DurationMarker).fold(input.readSimple().readLong())(_.toMillis).millis,
+    (output, value) => output.writeSimple().writeLong(value.toMillis),
+  )
+
+  implicit final val sizeInBytesCodec: GenCodec[SizeInBytes] = GenCodec.nonNull(
+    input => SizeInBytes(input.readCustom(SizeInBytesMarker).getOrElse(input.readSimple().readLong())),
+    (output, value) => output.writeSimple().writeLong(value.bytes),
+  )
+
+  implicit final val classKeyCodec: GenKeyCodec[Class[?]] =
+    GenKeyCodec.create(Class.forName, _.getName)
+
+  implicit final val classCodec: GenCodec[Class[?]] =
+    GenCodec.nullableString(Class.forName, _.getName)
```

Comment:
What would be the use case for those in config?

Reply:
We have a use case for entity class to database collection settings Map

Reply:
I won't die on this hill - this use case seems unrelated to you using HOCON, but if it helps a lot, it's rather harmless.